platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,277 @@
+of the event (its transactions) are limited in size and count, Event Intake will also check this as a safety measure in
+the event of bugs (if this check fails here, a noisy log statement should be produced, since this should never happen).
+
+If an event is valid, then we finally check the signature. Since validation and deduplication and hashing are
+significantly less expensive than signature verification, we wait on signature verification until the other steps are
+completed. The operating principle is that we want to fail fast and limit work for further stages in the pipeline.
+
+If an event has a very old birth-round that is expired, it is dropped. If a node sends a large number of expired events,
+it may end up being disciplined (the exact rules around this will be defined in subsequent design docs for the
+Event Intake module).
+
+#### Self Events
+
+Events are not only given to the event intake system through gossip. Internal events are also fed to the event intake
+system. These internal events **may** bypass some steps in the pipeline. For example, internal self-events (those
+events created by the node itself) do not need validation. Likewise, when replaying events from the pre-consensus
+recording system, those checks are not needed (since they have already been proved valid and are in topological order).
+
+#### Peer Discipline
+
+During the validation process, if an event is invalid, it is rejected, and this information is passed to the Bad Node
+module so the offending node may be disciplined. Note that the node to be disciplined will be the node that sent this
+bad event to us, not the origin node. This information (which node sent the event) must be captured by Gossip and
+passed to Event Intake as part of the event metadata.
+
+#### Topological Ordering
+
+Events are buffered if necessary to ensure that each parent event has been emitted from the Event Intake before any
+child events. A simple map (the same used for deduplication) can be used here. Given some event, for each parent, look
+up the parent by its hash. If each parent is found in the map, then emit the event. Otherwise, remember the event so
+when the missing parent is received, the child may be emitted. The current implementation uses what is known as the
+"orphan buffer" for this purpose.
+
+Since Event Intake will also maintain some buffers, it needs to know about the progression of the hashgraph,
+so it can evict old events. In this case, the "orphan buffer" holds events until either the parent events have
+arrived, or the events expired due to the advancement of the "non-ancient event window" and the event is
+dropped from the buffer. This document does not prescribe the existence of the orphan buffer or the method by which
+events are sorted and emitted in topological order, but it does describe a method by which old events can be dropped.
+
+#### Emitting Events
+
+When the Event Intake module emits valid, topologically sorted events, it sends them to:
```

Comment:
Upon further consideration, it seems to me that it would make more sense to delay sending an event to the event creator or "pre-handle", until *after* making the event durable in the PCES. This way, we can operate under the assumption that everything is durable.

Do we gain something by sending to these endpoints directly from intake, rather than waiting the (allegedly) insignificant amount of time for PCES durability?